datadog/__init__.py

```diff
@@ -80,7 +87,11 @@ def initialize(api_key=None, app_key=None, host_name=None, api_host=None,
         api._application_key or
         os.environ.get('DATADOG_APP_KEY', os.environ.get('DD_APP_KEY'))
     )
-    api._host_name = host_name or api._host_name or get_hostname()
```

Comment:
I actually wanted to keep this line that you removed, but it would call `get_hostname(hostname_from_config)` instead of just `get_hostname()`. We can't just ignore the `get_hostname` invocation completely, as it still does different things when the config is not found.